### PR TITLE
[WIP] NSB Core bumped up to 6.3.1

### PR DESCRIPTION
**Pull request by Scooletz**
Connects to #143 

Additionally, a lot of tests needed to be copied from the Acceptance Test suite as the API has changed. They need to be reviewed in depth. Also, should we try to use the storage only tests now?

### CODE REVIEW

**Comment by Scooletz** [comment]
Ping @Particular/azure-maintainers I want to discuss if we want to proceed with this PR. Leaving it as a WIP meanwhile.

File: src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_sending_from_a_saga_handle.cs [hunk 1]
```diff
@@ -40,6 +39,7 @@ public class TwoSaga1Saga1 : Saga<TwoSaga1Saga1Data>, IAmStartedByMessages<Start
             {
                 public Task Handle(StartSaga1 message, IMessageHandlerContext context)
                 {
+                    Data.DataId = message.DataId;
```
Comment (danielmarbach) [inline src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_sending_from_a_saga_handle.cs]: I thought the correlated property is set automatically

File: src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_sending_from_a_saga_handle.cs [hunk 2]
```diff
@@ -73,6 +73,7 @@ public class TwoSaga1Saga2 : Saga<TwoSaga1Saga2.TwoSaga1Saga2Data>, IAmStartedBy
 
                 public Task Handle(StartSaga2 message, IMessageHandlerContext context)
                 {
+                    Data.DataId = message.DataId;
```
Comment (danielmarbach) [inline src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_sending_from_a_saga_handle.cs]: I thought the correlated property is set automatically

File: src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_sending_from_a_saga_timeout.cs [hunk 3]
```diff
@@ -41,6 +43,7 @@ public class SendFromTimeoutSaga1 : Saga<SendFromTimeoutSaga1.SendFromTimeoutSag
 
                 public Task Handle(StartSaga1 message, IMessageHandlerContext context)
                 {
+                    Data.DataId = message.DataId;
```
Comment (danielmarbach) [inline src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_sending_from_a_saga_timeout.cs]: I thought the correlated property is set automatically

File: src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_sending_from_a_saga_timeout.cs [hunk 4]
```diff
@@ -70,31 +73,28 @@ public class SendFromTimeoutSaga2 : Saga<SendFromTimeoutSaga2.SendFromTimeoutSag
 
                 public Task Handle(StartSaga2 message, IMessageHandlerContext context)
                 {
+                    Data.DataId = message.DataId;
```
Comment (danielmarbach) [inline src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_sending_from_a_saga_timeout.cs]: I thought the correlated property is set automatically

File: src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_started_by_event_from_another_saga.cs [hunk 5]
```diff
@@ -65,6 +63,8 @@ public class EventFromOtherSaga1 : Saga<EventFromOtherSaga1.EventFromOtherSaga1D
 
                 public async Task Handle(StartSaga message, IMessageHandlerContext context)
                 {
+                    Data.DataId = message.DataId;
```
Comment (danielmarbach) [inline src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_started_by_event_from_another_saga.cs]: I thought the correlated property is set automatically

File: src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_started_by_event_from_another_saga.cs [hunk 6]
```diff
@@ -116,6 +115,7 @@ public class EventFromOtherSaga2 : Saga<EventFromOtherSaga2.EventFromOtherSaga2D
 
                 public Task Handle(SomethingHappenedEvent message, IMessageHandlerContext context)
                 {
+                    Data.DataId = message.DataId;
```
Comment (danielmarbach) [inline src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_started_by_event_from_another_saga.cs]: Same here

File: src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_timeout_hit_not_found_saga.cs [hunk 7]
```diff
@@ -1,166 +1,113 @@
...
             {
                 public Context TestContext { get; set; }
 
-                public Task Handle(GroupPendingEvent message, IMessageHandlerContext context)
+                public async Task Handle(StartSaga message, IMessageHandlerContext context)
                 {
-                    Console.Out.WriteLine("Saga1 received GroupPendingEvent for RunId: {0}", message.DataId);
-                    return context.SendLocal(new CompleteSaga1Now { DataId = message.DataId });
-                }
+                    Data.DataId = message.DataId;
```
Comment (danielmarbach) [inline src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/Sagas/When_timeout_hit_not_found_saga.cs]: Same here

**Comment by Scooletz** [comment]
I'm closing this in favor of a next PR that will take a different approach on bumping the core.

**Pull request closed without merging**